sdk/core/azure-core/inc/azure/core/base64.hpp

```diff
@@ -38,6 +39,21 @@ namespace Azure { namespace Core {
      */
     static std::string Base64Encode(const std::vector<uint8_t>& data);
 
+    /**
+     * @brief Encodes the vector of binary data into UTF-8 encoded text represented as Base64.
+     *
+     * @param data The binary data to be encoded.
+     * @param length The length of the binaryData parameter.
+     * @return The UTF-8 encoded text in Base64.
+     */
+    static std::string Base64Encode(uint8_t const* const data, size_t length);
+
+    template <size_t length>
```

Comment:
curious, why template?

Reply:
That allows me to specify an arbitrary std::array<uint8_t, length> - otherwise I need a specialization for each length.